src/components/select/Select.js

```diff
@@ -249,8 +249,8 @@ export default class SelectComponent extends Field {
     }
     if (typeof data === 'string') {
       const selectData = this.selectData;
-      if (selectData) {
-        data = selectData;
+      if (selectData && selectData[value]) {
```

Comment:
What if this.selectData is a string?

Reply:
I can't see any cases when this.selectData can be a string. There is only one place in code where selectData changes and it can be only object. 
See please src/components/select/Select.js:1425 
If you know any cases when selectData can be a string, please tell me.